rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -509,521 +636,55 @@ namespace rsx
-							{ NV4097_SET_TRANSFORM_PROGRAM, 512 }
-						};
-
-						if (has_deferred_call)
-						{
-							//Hopefully this is skippable so the batch can keep growing
-							for (const auto &method : skippable_ranges)
-							{
-								if (reg < method.first)
-									continue;
-
-								if (reg - method.first < method.second)
-								{
-									//Safe to ignore if value has not changed
-									if (method_registers.test(reg, value))
-									{
-										execute_method_call = false;
-										flush_commands_flag = false;
-									}
-
-									break;
-								}
-							}
-						}
-
-						break;
-					}
-					}
-
-					if (flush_commands_flag)
-					{
-						flush_command_queue();
-					}
-				}
-
-				if (capture_current_frame)
-				{
-					frame_debug.command_queue.push_back(std::make_pair(reg, value));
-
-					if (!(reg == NV406E_SET_REFERENCE || reg == NV406E_SEMAPHORE_RELEASE || reg == NV406E_SEMAPHORE_ACQUIRE))
-					{
-						// todo: handle nv406e methods better?, do we care about call/jumps?
-						rsx::frame_capture_data::replay_command replay_cmd;
-						replay_cmd.rsx_command = std::make_pair(i == 0 ? cmd : 0, value);
-
-						frame_capture.replay_commands.push_back(replay_cmd);
-
-						// to make this easier, use the replay command 'i' positions back
-						auto it = std::prev(frame_capture.replay_commands.end(), i + 1);
-
-						switch (reg)
-						{
-						case NV4097_GET_REPORT:
-							capture::capture_get_report(this, *it, value);
-							break;
-						case NV3089_IMAGE_IN:
-							capture::capture_image_in(this, *it);
-							break;
-						case NV0039_BUFFER_NOTIFY:
-							capture::capture_buffer_notify(this, *it);
-							break;
-						case NV4097_CLEAR_SURFACE:
-							capture::capture_surface_state(this, *it);
-							break;
-						default:
-							if (reg >= NV308A_COLOR && reg < NV3089_SET_OBJECT)
-								capture::capture_inline_transfer(this, *it, reg - NV308A_COLOR, value);
-							break;
-						}
-					}
-				}
+			// Execute backend-local tasks first
+			do_local_task(performance_counters.state);
 
-				method_registers.decode(reg, value);
+			// Update sub-units
+			zcull_ctrl->update(this);
 
-				if (execute_method_call)
-				{
-					if (auto method = methods[reg])
-					{
-						method(this, reg, value);
-					}
-				}
+			// Execute FIFO queue
+			run_FIFO();
+		}
+	}
 
-				if (invalid_command_interrupt_raised)
-				{
-					invalid_command_interrupt_raised = false;
+	void thread::on_exit()
+	{
+		m_rsx_thread_exiting = true;
 
-					//Skip the rest of this command
-					break;
-				}
-			}
+		if (m_vblank_thread)
+		{
+			m_vblank_thread->join();
+			m_vblank_thread.reset();
+		}
 
-			internal_get += (count + 1) * 4;
+		if (m_decompiler_thread)
+		{
+			m_decompiler_thread->join();
+			m_decompiler_thread.reset();
 		}
 	}
 
-	void thread::on_exit()
+	std::string thread::get_name() const
 	{
-		m_rsx_thread_exiting = true;
+		return "rsx::thread";
+>>>>>>> rsx: Fixups
```

Comment:
Remove this.

Reply:
Yes, didn't catch that (I just grep for <<<<< and forgot the opposite. Looks like some definitions were moved on master.

Reply:
Ahh, the threading system was refactored recently. These variables no longer exist.